File: 006/Ruby/problem006_test.rb
require "test/unit"

require_relative "problem006"

class TestProblem006 < Test::Unit::TestCase

  def test_diff_squares_sums
    assert_equal 0, 1.diff_squares_sums
    assert_equal 4, 2.diff_squares_sums
    assert_equal 22, 3.diff_squares_sums
    assert_equal 170, 5.diff_squares_sums
    assert_equal 2640, 10.diff_squares_sums
  end

  def test_sum_of_squares
    assert_equal 1, 1.sum_of_squares
    assert_equal 5, 2.sum_of_squares
    assert_equal 14, 3.sum_of_squares
    assert_equal 55, 5.sum_of_squares
  end

  def test_square_of_sum
    assert_equal 1, 1.square_of_sum
    assert_equal 9, 2.square_of_sum
    assert_equal 36, 3.square_of_sum
    assert_equal 225, 5.square_of_sum
  end

end

File: 006/Haskell/Problem006Test.hs
module Main where

import Test.HUnit
import System.Exit
import Problem006

diff_squares_sumsTests = TestList [
  TestCase (0 @=? diff_squares_sums 1),
  TestCase (4 @=? diff_squares_sums 2),
  TestCase (22 @=? diff_squares_sums 3),
  TestCase (170 @=? diff_squares_sums 5),
  TestCase (2640 @=? diff_squares_sums 10),
  TestCase (1 @=? 1)
  ]

sum_of_squaresTests = TestList [
  TestCase (1 @=? sum_of_squares 1),
  TestCase (5 @=? sum_of_squares 2),
  TestCase (14 @=? sum_of_squares 3),
  TestCase (55 @=? sum_of_squares 5),
  TestCase (1 @=? 1)
  ]

square_of_sumTests = TestList [
  TestCase (1 @=? square_of_sum 1),
  TestCase (9 @=? square_of_sum 2),
  TestCase (36 @=? square_of_sum 3),
  TestCase (225 @=? square_of_sum 5),
  TestCase (1 @=? 1)
  ]

tests = TestList [
	TestLabel "testing diff_squares_sums" diff_squares_sumsTests,
	TestLabel "testing sum_of_squares" sum_of_squaresTests,
	TestLabel "testing square_of_sum" square_of_sumTests,
	TestCase (1 @=? 1)
	]

main = runTestTT tests >>= checkTests
  where
    checkTests counts = case counts of
    	Counts _ _ errors failures ->
    		if (errors + failures > 0)
    		then exitFailure
    		else (exitWith ExitSuccess)
